classes/output/report_table.php

```diff
@@ -0,0 +1,180 @@
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * @package     local_smartmedia
+ * @author      Tom Dickman <tomdickman@catalyst-au.net>
+ * @copyright   2019 Catalyst IT Australia {@link http://www.catalyst-au.net}
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace local_smartmedia\output;
+
+defined('MOODLE_INTERNAL') || die;
+
+use table_sql;
+use moodle_url;
+use renderable;
+use html_writer;
+
+/**
+ * Renderable class for index page of local_smartmedia plugin.
+ *
+ * @package     local_smartmedia
+ * @author      Tom Dickman <tomdickman@catalyst-au.net>
+ * @copyright   2019 Catalyst IT Australia {@link http://www.catalyst-au.net}
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+class report_table extends table_sql implements renderable {
+
+    /**
+     * Minimum width above which transcoding is considered HD.
+     */
+    const MIN_HD_WIDTH = 720;
+
+    /**
+     * report_table constructor.
+     *
+     * @param string $uniqueid Unique id of table.
+     * @param \moodle_url $url Url where this table is displayed.
+     * @param \stdClass $params the parameters for table display.
+     *
+     * @throws \coding_exception
+     */
+    public function __construct($uniqueid, $url, $params) {
+        parent::__construct($uniqueid);
+
+        $this->set_attribute('id', 'local_smartmedia_report_table');
+        $this->set_attribute('class', 'generaltable generalbox');
+        $this->show_download_buttons_at(array(TABLE_P_BOTTOM));
+        $this->is_downloading($params->download, 'smartmedia-report');
+        $this->define_baseurl($url);
+        $this->define_columns(array('type', 'format', 'width', 'duration', 'size', 'cost'));
+        $this->define_headers(array(
+            get_string('report:type', 'local_smartmedia'),
+            get_string('report:format', 'local_smartmedia'),
+            get_string('report:resolution', 'local_smartmedia'),
+            get_string('report:duration', 'local_smartmedia'),
+            get_string('report:size', 'local_smartmedia'),
+            get_string('report:transcodecost', 'local_smartmedia')
+        ));
+        $this->currpage = isset($params->page) ? $params->page : 0;
+        $this->pagesize = isset($params->pagesize) ? $params->pagesize : 0;
+        $this->sortable(true);
+        $this->no_sorting('type');
+        $this->no_sorting('format');
+        $this->no_sorting('cost');
+        $this->set_sql('*', '{local_smartmedia_data}', 'TRUE');
```

Comment:
Lets only get the fields we are using explicitly.  No point getting them all if we are not using them all. However this is fine if we are getting them all

Reply:
Refactored to only use explicitly required fields.